### PR TITLE
chore(use-cases): update to aws sdk v3

### PR DESCRIPTION
**Pull request by biffgaut**

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: df48561b28abd19f1143b8613f6f56f93e8c5532
* Result: SUCCEEDED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b41da667-c726-492c-a90f-478d75e1b7a5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by hayesry — DISMISSED** [review]
LGTM

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 568b362631c056376478b32cd05b7eb7ff9bb11b
* Result: SUCCEEDED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6322dbde-1772-4ffe-b2be-6ff4e5457128%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by trivikr** [comment]
typo: The `vv2` in the title should be `v3`

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/kitchen-staff/get-open-orders/index.js [hunk 1]
```diff
@@ -12,8 +12,12 @@
  */
 
 // Imports
-const aws = require('aws-sdk');
-const ddb = new aws.DynamoDB.DocumentClient({apiVersion: '2012-08-10'});
+
+
+const { DynamoDBDocument } = require("@aws-sdk/lib-dynamodb");
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/kitchen-staff/get-open-orders/index.js:17]:
```suggestion
const { DynamoDBDocument } = require("@aws-sdk/lib-dynamodb");
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/layer/db-access.js [hunk 2]
```diff
@@ -12,8 +12,12 @@
  */
 
 // Imports
-const aws = require('aws-sdk');
-const ddb = new aws.DynamoDB.DocumentClient({apiVersion: '2012-08-10'});
+
+
+const { DynamoDBDocument } = require('@aws-sdk/lib-dynamodb');
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/layer/db-access.js:17]:
```suggestion
const { DynamoDBDocument } = require('@aws-sdk/lib-dynamodb');
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/archive-orders/index.js [hunk 3]
```diff
@@ -12,9 +12,14 @@
  */
 
 // Imports
-const aws = require('aws-sdk');
-const s3 = new aws.S3();
-const ddb = new aws.DynamoDB.DocumentClient({apiVersion: '2012-08-10'});
+
+
+const { DynamoDBDocument, DeleteCommand } = require('@aws-sdk/lib-dynamodb');
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/archive-orders/index.js:17]:
```suggestion
const { DynamoDBDocument, DeleteCommand } = require('@aws-sdk/lib-dynamodb');
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/kitchen-staff/complete-order/index.js [hunk 4]
```diff
@@ -12,8 +12,12 @@
  */
 
 // Imports
-const aws = require('aws-sdk');
-const ddb = new aws.DynamoDB.DocumentClient({apiVersion: '2012-08-10'});
+
+
+const { DynamoDBDocument } = require('@aws-sdk/lib-dynamodb');
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/kitchen-staff/complete-order/index.js:17]:
```suggestion
const { DynamoDBDocument } = require('@aws-sdk/lib-dynamodb');
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/calculate-tips/index.js [hunk 5]
```diff
@@ -12,8 +12,11 @@
  */
 
 // Imports
-const aws = require('aws-sdk');
-const sns = new aws.SNS();
+
+
+const { SNS, PublishCommand } = require('@aws-sdk/client-sns');
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/calculate-tips/index.js:17]:
```suggestion
const { SNS, PublishCommand } = require('@aws-sdk/client-sns');
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/check-late-orders/index.js [hunk 6]
```diff
@@ -12,9 +12,15 @@
  */
 
 // Imports
-const aws = require('aws-sdk');
-const ddb = new aws.DynamoDB.DocumentClient({apiVersion: '2012-08-10'});
-const sns = new aws.SNS();
+
+
+const { DynamoDBDocument } = require("@aws-sdk/lib-dynamodb");
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/check-late-orders/index.js:17]:
```suggestion
const { DynamoDBDocument } = require("@aws-sdk/lib-dynamodb");
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/close-out-service/index.js [hunk 7]
```diff
@@ -12,8 +12,11 @@
  */
 
 // Imports
-const aws = require('aws-sdk');
-const sfn = new aws.StepFunctions();
+
+
+const { SFN } = require('@aws-sdk/client-sfn');
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/close-out-service/index.js:17]:
```suggestion
const { SFN } = require('@aws-sdk/client-sfn');
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/create-report/index.js [hunk 8]
```diff
@@ -12,8 +12,11 @@
  */
 
 // Imports
-const aws = require('aws-sdk');
-const s3 = new aws.S3();
+
+
+const { S3 } = require('@aws-sdk/client-s3');
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/create-report/index.js:17]:
```suggestion
const { S3 } = require('@aws-sdk/client-s3');
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/get-report/index.js [hunk 9]
```diff
@@ -12,8 +12,11 @@
  */
 
 // Imports
-const aws = require('aws-sdk');
-const s3 = new aws.S3();
+
+
+const { S3 } = require('@aws-sdk/client-s3');
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/get-report/index.js:17]:
```suggestion
const { S3 } = require('@aws-sdk/client-s3');
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/service-staff/create-order/index.js [hunk 10]
```diff
@@ -12,15 +12,20 @@
  */
 
 // Imports
-const aws = require('aws-sdk');
-const ddb = new aws.DynamoDB.DocumentClient({apiVersion: '2012-08-10'});
+
+
+const { DynamoDBDocument } = require('@aws-sdk/lib-dynamodb');
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/service-staff/create-order/index.js:17]:
```suggestion
const { DynamoDBDocument } = require('@aws-sdk/lib-dynamodb');
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/service-staff/process-payment/index.js [hunk 11]
```diff
@@ -12,8 +12,12 @@
  */
 
 // Imports
-const aws = require('aws-sdk');
-const ddb = new aws.DynamoDB.DocumentClient({apiVersion: '2012-08-10'});
+
+
+const { DynamoDBDocument } = require("@aws-sdk/lib-dynamodb");
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/service-staff/process-payment/index.js:17]:
```suggestion
const { DynamoDBDocument } = require("@aws-sdk/lib-dynamodb");
```

**Review by trivikr — COMMENTED** [review]
nit: There's a minor issue in one of dependencies of aws-sdk-js-codemod where it adds additional newlines when there are comments.

These newlines can be removed. I've posted optional suggestions for all in this review, which can be applied through GitHub UI.

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/kitchen-staff/complete-order/index.js [hunk 12]
```diff
@@ -12,8 +12,12 @@
...
+const { DynamoDB } = require('@aws-sdk/client-dynamodb');
+
+const ddb = DynamoDBDocument.from(new DynamoDB({apiVersion: '2012-08-10'}));
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/kitchen-staff/complete-order/index.js:20]:
```suggestion
const ddb = DynamoDBDocument.from(new DynamoDB());
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/kitchen-staff/get-open-orders/index.js [hunk 13]
```diff
@@ -12,8 +12,12 @@
...
+const { DynamoDB } = require("@aws-sdk/client-dynamodb");
+
+const ddb = DynamoDBDocument.from(new DynamoDB({apiVersion: '2012-08-10'}));
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/kitchen-staff/get-open-orders/index.js:20]:
```suggestion
const ddb = DynamoDBDocument.from(new DynamoDB());
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/layer/db-access.js [hunk 14]
```diff
@@ -12,8 +12,12 @@
...
+const { DynamoDB } = require('@aws-sdk/client-dynamodb');
+
+const ddb = DynamoDBDocument.from(new DynamoDB({apiVersion: '2012-08-10'}));
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/layer/db-access.js:20]:
```suggestion
const ddb = DynamoDBDocument.from(new DynamoDB());
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/archive-orders/index.js [hunk 15]
```diff
@@ -12,9 +12,14 @@
...
+const { DynamoDB } = require('@aws-sdk/client-dynamodb');
+const { S3 } = require('@aws-sdk/client-s3');
+
+const s3 = new S3();
+const ddb = DynamoDBDocument.from(new DynamoDB({apiVersion: '2012-08-10'}));
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/archive-orders/index.js:22]:
```suggestion
const ddb = DynamoDBDocument.from(new DynamoDB());
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/check-late-orders/index.js [hunk 16]
```diff
@@ -12,9 +12,15 @@
...
+const { DynamoDB } = require("@aws-sdk/client-dynamodb");
+const { SNS } = require("@aws-sdk/client-sns");
+const { PublishCommand } = require("@aws-sdk/client-sns");
+
+const ddb = DynamoDBDocument.from(new DynamoDB({apiVersion: '2012-08-10'}));
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/check-late-orders/index.js:22]:
```suggestion
const ddb = DynamoDBDocument.from(new DynamoDB());
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/service-staff/create-order/index.js [hunk 17]
```diff
@@ -12,15 +12,20 @@
...
+const { DynamoDB } = require('@aws-sdk/client-dynamodb');
+
+const ddb = DynamoDBDocument.from(new DynamoDB({apiVersion: '2012-08-10'}));
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/service-staff/create-order/index.js:20]:
```suggestion
const ddb = DynamoDBDocument.from(new DynamoDB());
```

**Review by trivikr — COMMENTED** [review]
nit: The `apiVersion` is no longer honored by JS SDK v3 and can be safely.
The aws-sdk-js-codemod adds a comment suggesting removal https://github.com/aws/aws-sdk-js-codemod/blob/8f023db7ce8ad544b07bd3948082c8d53befbfed/src/transforms/v2-to-v3/__fixtures__/config/apiVersion.output.js#L2-L4

I've posted optional suggestions for removal in this review, which can be applied through GitHub UI.

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/archive-orders/index.js [hunk 18]
```diff
@@ -56,31 +61,21 @@ exports.handler = async (event) => {
...
+    console.log(`deleting: ${entry.id}`);
     // Delete the entry
-    ddb.delete(params, function(err, data) {
-      if (err) console.log(err);
-      else {
-        console.log(data);
-        deleteEntries(entries);
-      }
-    });
+    await ddb.send(new DeleteCommand(params));
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/archive-orders/index.js:79]:
```suggestion
    await ddb.delete(params);
```

File: source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/calculate-tips/index.js [hunk 19]
```diff
@@ -35,25 +38,21 @@ exports.handler = async (event) => {
...
       TopicArn: process.env.SNS_TOPIC_ARN
     };
-    console.log(sns_params);
-    // Send the message
-    sns.publish(sns_params, function(err, data) {
-      if (err) console.log(err, err.stack); // an error occurred
-      else     console.log(data);           // successful response
-    });
-  })
+    await sns.send(new PublishCommand(sns_params));
```
Comment (trivikr) [inline source/use_cases/aws-restaurant-management-demo/lib/lambda/manager/calculate-tips/index.js:55]:
```suggestion
    await sns.publish(sns_params);
```

**Review by trivikr — COMMENTED** [review]
nit: since rest of the code makes direct API calls instead of using command objects, the two lines which are different can do the same.

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 9866dcb2bc3beaa8ce4a5d36bd5a0e8db2647eac
* Result: SUCCEEDED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a038f6eb-d16e-4e6a-8968-1c692e965f68%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 5b531a6a3f16610457ac2d55b1f5470a1ed586b1
* Result: SUCCEEDED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=90dde6d8-972d-49a3-a31b-5b27adbe7c70%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 8f52fddccc696cdffa5269e9ba2718aec201e627
* Result: SUCCEEDED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b0097ef6-23ec-4d55-8ac0-9015ab6cc3f8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: cafbd04ccb7821a85a8c7b027c7260d392bc61f5
* Result: SUCCEEDED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2b81a131-eb61-412c-8dbc-293d8ca966d7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 1a9cf26e4215a3fbdcfbf5cf704da7faa588f913
* Result: SUCCEEDED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=31b2f180-97eb-48c9-9ea6-141d0599c83e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by hayesry — APPROVED** [review]
No concerns

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 9ccc4dd4c34230490c8cccddbc5f8f809cffda79
* Result: SUCCEEDED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3bdcc427-51d5-4662-b9e6-01b9f2db846e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**